### ISSUE TITLE
Khal randomly stopped working 'No module named 'xdg.BaseDirectory'

### ISSUE THREAD

**Issue by AtomToast** [pimutils/khal#959]
Up until yesterday (2020-09-17) khal worked perfectly fine. Then I tried to comment out one value in my config and ever since then khal crashes everytime with

```
Traceback (most recent call last):
  File "/usr/bin/khal", line 2, in <module>
    from khal.cli import main_khal
  File "/usr/lib/python3.8/site-packages/khal/cli.py", line 35, in <module>
    from .settings import InvalidSettingsError, get_config
  File "/usr/lib/python3.8/site-packages/khal/settings/__init__.py", line 1, in <module>
    from .settings import get_config  # noqa
  File "/usr/lib/python3.8/site-packages/khal/settings/settings.py", line 26, in <module>
    import xdg.BaseDirectory
ModuleNotFoundError: No module named 'xdg.BaseDirectory'
```

Even after undoing the change. I can delete the whole config directory and the error still persists.

I also tried reinstalling khal and all it's dependencies but still no luck. I really have no clue why this is happening. Any tips for troubleshooting would be greatly appreciated

System is Arch Linux, latest updates at the time of posting this

**Comment by d7415** [comment]
It looks like you need python3-xdg (or whatever Arch calls it).

**Comment by AtomToast** [comment]
Yeah, that's what I thought as well but it was installed and I even also reinstalled it and it didn't help

**Comment by d7415** [comment]
Ok. Well khal can't see it. How did you install khal and how did you install python3-xdg?

**Comment by AtomToast** [comment]
It could see it before though. I installed both from the official repos

**Comment by WhyNotHugo** [comment]
That's very odd. I'm also using both packages from the arch repos and it all works for me. What's the output of `pacman -Ql python-xdg`, of `python -c "import xdg` and of `which Python`.

**Comment by AtomToast** [comment]
output of `pacman -Ql python-xdg`: https://termbin.com/uvrtd

`python -c "import xdg"` doesn't return anything but exits successfully

`which python` returns `/usr/bin/python`

I can't even run khal --version but from pacman I know that khal is version `0.10.2-1`
python is version `3.8.5`
python-xdg is version `0.26-6`

**Comment by WhyNotHugo** [comment]
Darn, all that looks in order.

What's `which khal` and `head $(which khal)`?

**Comment by AtomToast** [comment]
`which khal` is `/usr/bin/khal`

`head $(which khal)` is

```py
#!/usr/bin/env python
from khal.cli import main_khal

if __name__ == "__main__":
    main_khal()
```

This whole thing is so bizarre

**Comment by d7415** [comment]
Hmm.. How about `/usr/bin/env python -c "import xdg"`?

**Comment by AtomToast** [comment]
No,works just fine.

However I just had an idea: I tried to instead of just importing xdg to import the thing khal failed at which was `import xdg.BaseDirectory`. That did error out

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'xdg.BaseDirectory'

```

Now even more confusingly from what I can see the xdg module does not seem to contain any trace of the BaseDirectory submodule https://github.com/srstevenson/xdg

**Comment by AtomToast** [comment]
I resolved the issue though I am not sure what caused it in the first place.

Instead of using the distros package I decided to try installing xdg via pip. However that told me that the dependency was already satisfied even though I uninstalled the repo package. It was pointing to a path at ~/.local/lib/python3.8/site-packages/

Looking there there were two directories named xdg. At first glance they did not seem to contain much so I just deleted them. Then, after installing python-pyxdg again, khal worked fine again.

I assume it tried to import the package from that directory which was somehow faulty but I don't why it did or why there was something in the first place. Reinstalling python-pyxdg did not recreate those folders.
